Sources/GRPCNIOTransportCore/Client/Connection/GRPCChannel.swift

```diff
@@ -758,14 +758,30 @@ extension GRPCChannel.StateMachine {
             result: .success(state.current)
           )
 
-        case .transientFailure, .shutdown:  // shutdown includes shutting down
+        case .transientFailure(let cause):
           // Current load-balancer failed. Remove all the 'fast-failing' continuations in the
           // queue, these are RPCs which set the 'wait for ready' option to false. The rest of
           // the entries in the queue will wait for a load-balancer to become ready.
           let continuations = state.queue.removeFastFailingEntries()
           actions.resumeContinuations = ConnectivityStateChangeActions.ResumableContinuations(
             continuations: continuations,
-            result: .failure(RPCError(code: .unavailable, message: "channel isn't ready"))
+            result: .failure(
+              RPCError(
+                code: .unavailable,
+                message: "Channel isn't ready.",
+                cause: cause
+              )
+            )
```

Comment:
IIUC the cause is already constrained to be an `RPCError` here – we should just use it rather than wrapping it.